### ISSUE TITLE
Add option to minimize generated html file

### ISSUE THREAD

**Issue by qarmin** [igiagkiozis/plotly#150]
Currently when using plot.to_html(), generated file is really big for big amount of data(my app can generate few plots with several thousands of points).

I found that removing unnecessary new lines and empty points can decrease size of plot by ~30%

I tried to use html/js minifier, but it doesn't work, so I wrote simple regex which do almost same thing
```
    let mut html = plot.to_html();
    let regex = Regex::new(r"\n[ ]+").unwrap();
    let html = regex.replace_all(&html, "");
```

**Comment by mfreeborn** [comment]
My inclination would be that minification would be outside the purview of this crate. There are many established minifiers available, and to bring it into the scope of this crate would just add dependencies and whilst providing a sub-optimal implementation.

When you say that you tried a minifier and it didn't work, can you expand on what happened?

**Comment by qarmin** [comment]
I used https://github.com/wilsonzlin/minify-html, but looks that js created by this library is unsupported yet or non standard.

Code was similar to
```
use minify_html::{Cfg, minify};

let mut code: &[u8] = html.as_bytes();
let mut cfg = Cfg::new();
cfg.minify_js = true;
let minified = minify(&code, &cfg);
```

**Comment by andrei-ng** [comment]
I agree with @mfreeborn . Minifying is out of scope of this crate. Since this issue was raised, embedding plotly.js has also been put behind a feature flag, hence using CDN versions should also result in smaller html files
